pages/help/services-resources/index.vue

```diff
@@ -48,29 +36,101 @@ if (!data.value.data && !data.value.helpTopicData) {
   throw createError({ statusCode: 404, message: 'Page not found', fatal: true })
 }
 
+// ELASTIC SEARCH INDEX
+// GETS DATA FROM CRAFT
+// CREATES ES INDEX TO BE SEARCHED
+// CHECK THAT NUXT IS RUNNING ON THE SERVER (process.server)
+console.log('DATA-DATA-DATA-DATA' + data)
+if (
+  data.value.externalResource &&
+  data.value.externalResource.length > 0 &&
+  process.server
+) {
+  for (const externalResource of data.value.externalResource) {
+    await $elasticsearchplugin.index(
+      { ...externalResource, serviceOrResourceType: 'external resource' },
+      externalResource.slug
+    )
+  }
+}
+
 const page = ref(data.value.data)
 const helpTopic = ref(data.value.helpTopicData)
 const summaryData = ref(_get(data.value.data, 'entry', {}))
+
+// ENABLE PREVIEW
 watch(data, (newVal, oldVal) => {
   console.log('In watch preview enabled, newVal, oldVal', newVal, oldVal)
   page.value = _get(newVal.data, 'entry', {})
   helpTopic.value = ref(newVal.helpTopicData)
   summaryData.value = ref(_get(newVal.data, 'entry', {}))
 })
 
+// FUNCTION AND METHODS RELATED TO ES
+// COMPUTED METHODS THAT USE CRAFT DATA
+const hits = ref([])
 const noResultsFound = ref(false)
 const searchFilters = ref([])
-const hits = ref([])
-const searchGenericQuery = ref(
-  {
-    queryText: route.query.q || '',
-    queryFilters:
+
+const searchGenericQuery = ref({
+  queryText: route.query.q || '',
+  queryFilters:
+    (route.query.filters &&
+      JSON.parse(route.query.filters)) ||
+    {},
+})
+
+// THIS WATCHER IS CALLED WHEN THE ROUTER PUSHES UPDATES TO THE QUERY PARAM
+// ie: someone changes the query or starts viewing from a bookmarked page
+watch(() => route?.query, (oldValue, newValue) => {
+  if (oldValue !== newValue) {
```

Comment:
Add a comment before this line 
`console.log('ES newVal, oldVal', newVal, oldVal)`
Looks like we copied this from access/collection, Also access/collection does not have fiters and only keyword text field,Also  in javascript we really cannot do oldValue !== newValue if the oldvalue and newValue are nested objects. One more thing this template has both query.q and query.filters
So replace
`if (oldValue !== newValue) {
    if (newValue?.q === '') hits.value = []
    searchGenericQuery.value.queryText = route.query.q || ''
    searchES()
  }`
  
with
`
    searchGenericQuery.value.queryText = route.query.q || ''
    searchGenericQuery.value.queryFilters = (route.query.filters && JSON.parse(route.query.filters)) || {}
    searchES()
  `
  Look at news listing or events and exhibition listing page for similar examples with filters